### ISSUE TITLE
Error in build boost c++ library during installing Dyninst in Linux Ubuntu

### ISSUE THREAD

**Issue by 1Husin-Alhaj** [dyninst/dyninst#435]
After spending many hours and days on the installation of the Dynins api tool, all attempts failed. I have tried my best to raise this problem but I could not. Please help me solve this problem, because I desperately need Dyninst to complete work.

Here is the error ... 

husien@husien-F52Q:~/Documents/cmake-build$ sudo make
[  0%] Building DyninstRT
[ 50%] Built target dyninstAPI_RT
[100%] Built target dyninstAPI_RT_static
[  0%] Built target DyninstRT
[  2%] Built target LibElf
[  3%] Built target LibIberty
[  3%] Performing download step (download, verify and extract) for 'boost'
-- downloading...
     src='http://downloads.sourceforge.net/project/boost/boost/1.61.0/boost_1_61_0.zip'
     dst='/home/husien/Documents/cmake-build/boost/src/boost_1_61_0.zip'
     timeout='none'
CMake Error at boost-stamp/download-boost.cmake:9 (file):
  file DOWNLOAD HASH mismatch

    for file: [/home/husien/Documents/cmake-build/boost/src/boost_1_61_0.zip]
      expected hash: [015ae4afa6f3e597232bfe1dab949ace]
        actual hash: [d41d8cd98f00b204e9800998ecf8427e]
             status: [22;"HTTP response code said error"]

CMakeFiles/boost.dir/build.make:82: recipe for target 'boost/src/boost-stamp/boost-download' failed
make[2]: *** [boost/src/boost-stamp/boost-download] Error 1
CMakeFiles/Makefile2:165: recipe for target 'CMakeFiles/boost.dir/all' failed
make[1]: *** [CMakeFiles/boost.dir/all] Error 2
Makefile:117: recipe for target 'all' failed
make: *** [all] Error 2

I have installed the boost c++ library and re-installed Dyndns, but the error still occurs.

any help will be appreciated

Al-haj Ahmad

**Comment by mxz297** [comment]
@wrwilliams I am not quite familiar with our build system. Can you help and see what is going wrong?

**Comment by 1Husin-Alhaj** [comment]
Thank you very much Mr. Williams
I have read the installation file carefully and I have configured the
Dyninst with some extra parameters, the problem is solved.
I typed this command line in terminal:
sudo cmake /home/husien/Downloads/dyninst-master
-DBOOST_ROOT=/home/husien/Downloads/boost_1_66_0

I thing because my IP is from iran, during installing Dynins, boost c++
library is required. I have downloaded boost and installed it. Then During
installing Dynint, I added the configration parameter -DBOOST_ROOT.

Thanks
Al-haj Ahmad
